File: module-config/pull-with-push-override/mod/module0.go
/*
 *  Copyright © 2021 Paulo Villela. All rights reserved.
 *  Use of this source code is governed by the Apache 2.0 license
 *  that can be found in the LICENSE file.
 */

package mod

import (
	"fmt"
	"github.com/pvillela/go-tryout/module-config/pull-with-push-override/fwk"
)

type Module0CfgT struct {
	X string
}

var Module0Cfg = fwk.MakeConfigSource[Module0CfgT](Module0Adapter)

func Foo() {
	fmt.Println(Module0Cfg.Get().X)
	Bar()
}

File: module-config/push-to-function/module0.ts
/*
 * Copyright © 2022 Paulo Villela. All rights reserved.
 * Use of this source code is governed by the MIT license
 * that can be found in the LICENSE file.
 */

export type Cfg = {
  x string;
};

export type Dep = {
  bar () => void;
};

export type Foo = () => void;

func fooC(cfg () => Cfg, dep Dep) Foo {
  return function() {
    console.log(cfg().x);
    dep.bar()
  }
}
